### DISCUSSION TITLE
OPCUA Server redundancy

### DISCUSSION THREAD

**Discussion by vidyarao123** [FreeOpcUa/python-opcua#1260] [Q&A]
I have 2 OPCUA servers (in different machines) and 1 opcua client . The client will be connected to one server at a time. when the primary server fails, then the client must get connected to the secondary server automatically. 

Is this scenario possible to implement using python opcua module ??

**Comment by AndreasHeine** [comment]
sure!
https://reference.opcfoundation.org/v104/Core/docs/Part4/6.6.2/

**Reply by vidyarao123** [reply]
You mean through your python opcua module it is possible ?

**Reply by AndreasHeine** [reply]
sure! 
but you should read the link from the reference before continue asking questions its good explained :)
this behaivior must be implemented by yourself!

**Reply by vidyarao123** [reply]
Alright ,, Sorry I wanted to make sure if I have made my point clear,..

**Reply by AndreasHeine** [reply]
in opcua-asyncio its already implemented by @ZuZuD https://github.com/FreeOpcUa/opcua-asyncio/blob/master/asyncua/client/ha/ha_client.py but not in python-opcua (lib is in maintaince mode)
